app/com/arpnetworking/metrics/portal/scheduling/impl/PeriodicSchedule.java

```diff
@@ -32,26 +36,68 @@
 public final class PeriodicSchedule extends BaseSchedule {
 
     private final ChronoUnit _period;
+    private final ZoneId _zone;
     private final Duration _offset;
 
     private PeriodicSchedule(final Builder builder) {
         super(builder);
         _period = builder._period;
+        _zone = builder._zone;
         _offset = builder._offset;
     }
 
     @Override
-    protected Optional<ZonedDateTime> unboundedNextRun(final Optional<ZonedDateTime> lastRun) {
-        if (!lastRun.isPresent()) {
-            ZonedDateTime alignedStart = getRunAtAndAfter().truncatedTo(_period);
-            if (alignedStart.isBefore(getRunAtAndAfter())) {
-                alignedStart = alignedStart.plus(Duration.of(1, _period));
-            }
-            return Optional.of(alignedStart.plus(_offset));
+    protected Optional<Instant> unboundedNextRun(final Optional<Instant> lastRun) {
+        final Instant runAfter = lastRun.orElse(getRunAtAndAfter().minus(Duration.ofNanos(1)));
+        return Optional.of(nextAlignedBoundaryAfter(runAfter).plus(_offset));
+    }
+
+    /**
+     * Returns the first instant after the given boundary that is [period]-aligned in our time zone.
+     *
+     * @param boundary The exclusive lower bound.
+     * @return A boundary that is guaranteed to be [period]-aligned and after the given boundary.
+     */
+    private Instant nextAlignedBoundaryAfter(final Instant boundary) {
+        final ZonedDateTime zonedLastRun = ZonedDateTime.ofInstant(boundary, _zone);
+        final ZonedDateTime truncated = zonedLastRun.truncatedTo(_period);
+        ZonedDateTime result = truncated.plus(_period.getDuration()).truncatedTo(_period);
```

Comment:
This implementation seems rather complicated, and reading the comments I am unclear as to the cause. 

First, the caller, `unboundedNextRun`, takes the `runAtAndAfter` and substracts one nanosecond. Why is this done?

= First Run =
* Take runAtAndAfter and truncate to period
* Add period until the date is equal to or after runAtAndAfter (e.g. not before); this will be zero or one addition

= Next Run =
* Truncate the last run date to the period to remove the offset which must be non-negative and less than the period. Also accounts for any period changes since the last run.
* Take the truncated last run date and add the interval. 


After, to either of these results add the offset.